src/main/java/net/snowflake/client/jdbc/cloud/storage/StorageClientFactory.java

```diff
@@ -122,6 +122,13 @@ private SnowflakeS3Client createS3Client(
     clientConfig.setMaxErrorRetry(S3_TRANSFER_MAX_RETRIES);
     clientConfig.setDisableSocketProxy(HttpUtil.isSocksProxyDisabled());
 
+    // If proxy is set via connection properties or JVM settings these will be overridden later.
+    // This is to prevent the aws client builder from reading proxy environment variables.
+    clientConfig.setProxyHost("");
```

Comment:
will it disable AWS S3 default configuration? How about they can set proxy host but they wanted to use proxy password using environment variable? Why we enforce to use connection property or JVM setting? will it consider as BCR? will it cause any regression?

Reply:
@sfc-gh-igarish This is the way it's implemented in the driver: https://docs.snowflake.com/en/developer-guide/jdbc/jdbc-configure#connecting-using-a-proxy-server. The driver itself doesn't have the capability to read any proxy settings from environment variables including proxy password so they shouldn't be trying to do that anyway. [The default values for the driver proxy](https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java#L19) are the same as the change in this PR for the S3 proxy:

```
  private String proxyHost = "";
  private int proxyPort = 0;
  private String nonProxyHosts = "";
  private String proxyUser = "";
  private String proxyPassword = "";
```